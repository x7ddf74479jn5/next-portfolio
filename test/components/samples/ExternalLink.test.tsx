import { render } from "@testing-library/react";
import ExternalLink from "src/components/samples/ExternalLink";

describe("ExternalLink", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<ExternalLink label="test" url="test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    const { container, getByRole } = render(<ExternalLink label="test" url="test" />);

    const anchorElement = getByRole("link");
    expect(anchorElement).toHaveAttribute("href", expect.stringContaining("test"));
    expect(anchorElement).toHaveAttribute("target", "_blank");
    expect(anchorElement).toHaveAttribute("rel", "noreferrer");

    expect(container).toHaveTextContent("test");
  });
});
