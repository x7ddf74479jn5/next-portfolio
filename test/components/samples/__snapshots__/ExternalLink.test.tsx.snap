// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="test"
    rel="noreferrer"
    target="_blank"
  >
    <span
      class="externalLink"
    >
      test
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </span>
  </a>
</DocumentFragment>
`;
